src/hooks/useProductsList/useProductsList.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react'
+
+import open from '../../utils/indexedDB/indexedDB'
+import reducer, * as actions from './reducer'
+import initialState from './handlers'
+
+import useConfigAutosave from './utils/useConfigAutosave'
+import createRandomProducts from './utils/createRandomProducts'
+
+function useProductsList(user, draft) {
+  const [state, dispatch] = React.useReducer(reducer, initialState)
+  const database = React.useRef()
+
+  const DATABASE = {
+    name: user.company,
+    version: 1,
+    store: 'products',
+    key: 'id',
+  }
+
+  // NOTE: use of React.useCallback ?
+  const dispatchLoadList = (list) => dispatch(actions.loadList(list))
+  const dispatchAddItem = (item) => dispatch(actions.addItem(item))
+  const dispatchAddArray = (array) => dispatch(actions.addArray(array))
+  const dispatchDeleteItem = (id) => dispatch(actions.deleteItem(id))
+  const dispatchUpdateItem = (update) => dispatch(actions.updateItem(update))
+  const dispatchClearList = () => dispatch(actions.clearList())
+
+  function timedReset() {
+    window.setTimeout(function () {
+      dispatch(actions.requestReset())
+    }, 4000)
+  }
+
+  React.useEffect(() => {
+    open(DATABASE)
+      .then((dbRef) => (database.current = dbRef))
+      .then(loadList)
+      .catch(console.error)
+  }, [])
```

Comment:
Separate and name what this function is supposed to do  
See example in  `useAuthentication.js`